libraries/DataFlash/LogStructure.h

```diff
@@ -34,6 +36,91 @@ struct PACKED log_Format {
     char labels[64];
 };
 
+struct PACKED log_Unit {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    char unit[64]; // you know, this might be overkill...
+};
+
+struct PACKED log_Format_Multiplier {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    char type;
+    double multiplier;
+};
+
+struct PACKED log_Format_Units {
+    LOG_PACKET_HEADER;
+    uint64_t time_us;
+    uint8_t format_type;
+    char units[16];
+    char multipliers[16];
+};
+
+struct UnitStructure {
+    const char ID;
+    const char unit[64];
+};
+
+struct MultiplierStructure {
+    const char ID;
+    const double multiplier;
+};
+
+// all units here should be base units
+// This does mean battery capacity is here as "amps*seconds"
+const struct UnitStructure log_Units[] = {
+    { '-', "" },    // no units e.g. Pi, or a string
+    { '?', "UNKNOWN" },    // Units which haven't been worked out yet....
+    { 'A', "amps" },
+    { 'd', "degrees" }, // of the angular variety, -180 to 180
+    { 'D', "deglatitude" },
+    { 'b', "bytes" },
+    { 'k', "degrees/second" },
+    { 'e', "degrees/second/second" },
+    { 'E', "radians/second" },
+    { 'G', "gauss" },
+    { 'h', "degreesheading" }, // 0.? to 359.?
+    { 'i', "amps*seconds" },
+    { 'm', "metres" },
+    { 'n', "metres/second" },
+    // { 'N', "Newtons" },
+    { 'o', "metres/second/second" },
+    { 'O', "degcelsius" },
+    { 'S', "satellites" },
+    { 's', "seconds" },
+    { 'q', "rpm" },
+    { 'r', "radians" },
+    { 'u', "pulses/minute" }, // i.e. ppm
+    { 'U', "deglongitude" },
+    { 'v', "volts" },
+    { 'P', "pascals" },
+    { 'w', "ohms" },
+    { 'z', "hertz" }
+};
+
+// this multiplier information applies to the raw value present in the
+// log.  Any adjustment implied by the format field (e.g. the "centi"
+// in "centidegrees" is *IGNORED* for the purposes of scaling.
+// Essentially "format" simply tells you the C-type, and format-type h
+// (int16_t) is equivalent to format-type c (int16_t*100)
+const struct MultiplierStructure log_Multipliers[] = {
```

Comment:
Can these be emitted as floats instead? The burden on the GCS of having a LUT for this is a pain, and will result in more breakages, if we just wrote out the actual float's then we could handle arbitrary scaling as needed.

Reply:
I missed the MULT message, feel free to ignore the comment :)